mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,651 @@
+                y=self.y,
+                y_probs=self.y_prob,
+                labels=self.label_list,
+                curve_type="roc",
+            )
+
+            def plot_roc_curve():
+                roc_curve.plot_fn(**roc_curve.plot_fn_args)
+
+            self._log_image_artifact(plot_roc_curve, "roc_curve_plot")
+            self.metrics["roc_auc"] = roc_curve.auc
+
+            pr_curve = _gen_classifier_curve(
+                is_binomial=True,
+                y=self.y,
+                y_probs=self.y_prob,
+                labels=self.label_list,
+                curve_type="pr",
+            )
+
+            def plot_pr_curve():
+                pr_curve.plot_fn(**pr_curve.plot_fn_args)
+
+            self._log_image_artifact(plot_pr_curve, "precision_recall_curve_plot")
+            self.metrics["precision_recall_auc"] = pr_curve.auc
+
+    def _log_multiclass_classifier(self):
+        per_class_metrics_collection_df = _get_classifier_per_class_metrics_collection_df(
+            self.y, self.y_pred, self.label_list
+        )
+
+        log_roc_pr_curve = False
+        if self.y_probs is not None:
+            max_num_classes_for_logging_curve = self.evaluator_config.get(
+                "max_num_classes_threshold_logging_roc_pr_curve_for_multiclass_classifier", 10
+            )
+            if self.num_classes <= max_num_classes_for_logging_curve:
+                log_roc_pr_curve = True
+            else:
+                _logger.warning(
+                    f"The classifier num_classes > {max_num_classes_for_logging_curve}, skip "
+                    f"logging ROC curve and Precision-Recall curve. You can add evaluator config "
+                    f"'max_num_classes_threshold_logging_roc_pr_curve_for_multiclass_classifier' "
+                    f"to increase the threshold."
+                )
+
+        if log_roc_pr_curve:
+            roc_curve = _gen_classifier_curve(
+                is_binomial=False,
+                y=self.y,
+                y_probs=self.y_probs,
+                labels=self.label_list,
+                curve_type="roc",
+            )
+
+            def plot_roc_curve():
+                roc_curve.plot_fn(**roc_curve.plot_fn_args)
+
+            self._log_image_artifact(plot_roc_curve, "roc_curve_plot")
+            per_class_metrics_collection_df["roc_auc"] = roc_curve.auc
+
+            pr_curve = _gen_classifier_curve(
+                is_binomial=False,
+                y=self.y,
+                y_probs=self.y_probs,
+                labels=self.label_list,
+                curve_type="pr",
+            )
+
+            def plot_pr_curve():
+                pr_curve.plot_fn(**pr_curve.plot_fn_args)
+
+            self._log_image_artifact(plot_pr_curve, "precision_recall_curve_plot")
+            per_class_metrics_collection_df["precision_recall_auc"] = pr_curve.auc
+
+        self._log_pandas_df_artifact(per_class_metrics_collection_df, "per_class_metrics")
```

Comment:
This is a good design choice to store it as a .csv in the artifacts. Keeps the tracking metrics entries clean. +1